# HG changeset patch
# User Michael Godfrey <michaeldgodfrey@gmail.com>
# Date 1374790712 14400
#      Thu Jul 25 18:18:32 2013 -0400
# Node ID d3ebc3f9bdbffeb35b4dad502fdd49985589753f
# Parent  872947f3f9f16a2e6b21b5f96d72502571f962a5
doc update for grid and subplot

diff --git a/scripts/plot/grid.m b/scripts/plot/grid.m
--- a/scripts/plot/grid.m
+++ b/scripts/plot/grid.m
@@ -66,17 +66,17 @@ function grid (varargin)
               && strcmp (get (hax, "yminorgrid"), "on")
               && strcmp (get (hax, "zminorgrid"), "on"));
 
   if (nargs == 0)
     grid_on = ! grid_on;
   else
     x = varargin{1};
     if (! ischar (x))
-      error ("grid: argument must be a string");
+      error ("grid: argument 1 must be an axis handle or a string");
     endif
     if (strcmpi (x, "off"))
       grid_on = false;
     elseif (strcmpi (x, "on"))
       grid_on = true;
     elseif (strcmpi (x, "minor"))
       if (nargs == 2)
         x2 = varargin{2};
diff --git a/scripts/plot/subplot.m b/scripts/plot/subplot.m
--- a/scripts/plot/subplot.m
+++ b/scripts/plot/subplot.m
@@ -14,19 +14,23 @@
 ##
 ## You should have received a copy of the GNU General Public License
 ## along with Octave; see the file COPYING.  If not, see
 ## <http://www.gnu.org/licenses/>.
 
 ## -*- texinfo -*-
 ## @deftypefn  {Function File} {} subplot (@var{rows}, @var{cols}, @var{index})
 ## @deftypefnx {Function File} {} subplot (@var{rcn})
+## @deftypefnx {Function File} {@var{ax}} = subplot (@var{rcn}, ...)
 ## Set up a plot grid with @var{rows} by @var{cols} subwindows and plot
 ## in location given by @var{index}.
 ##
+## If @var{ax} is provided, subplot returns the axis handle for the subplot.
+## This is usuful for modifying the properties of a subplot.
+##
 ## If only one argument is supplied, then it must be a three digit value
 ## specifying the location in digits 1 (rows) and 2 (columns) and the plot
 ## index in digit 3.
 ##
 ## The plot index runs row-wise.  First all the columns in a row are filled
 ## and then the next row is filled.
 ##
 ## For example, a plot with 2 by 3 grid will have plot indices running as
